### PR TITLE
incremental: postbuild phase save intermediate files and relay messages

### PR DESCRIPTION
**Pull request by ansyral**
@vwxyzh @vicancy @superyyrrzz @hellosnow @qinezh @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 1]
```diff
@@ -102,42 +101,21 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
...
                 List<HostService> hostServices = null;
+                IHostServiceCreator hostServiceCreator = null;
+                PhaseProcessor phaseProcessor = null;
                 try
                 {
                     using (var templateProcessor = parameters.TemplateManager?.GetTemplateProcessor(context, parameters.MaxParallelism) ?? TemplateProcessor.DefaultProcessor)
                     {
+                        using (new LoggerPhaseScope("Prepare", true))
+                        {
+                            Prepare(parameters, context, templateProcessor, out hostServiceCreator, out phaseProcessor);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: >Prepare [](start = 28, length = 7)

what if Prepare fails, e.g. incremental files get deleted, fall back to clean build?

File: src/Microsoft.DocAsCode.Build.Engine/PostbuildPhaseHandler.cs [hunk 2]
```diff
@@ -9,11 +9,14 @@ namespace Microsoft.DocAsCode.Build.Engine
 
     internal class PostbuildPhaseHandler : IPhaseHandler
     {
-        private DocumentBuildContext _context;
+        public DocumentBuildContext Context { get; }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/PostbuildPhaseHandler.cs]: >Context [](start = 36, length = 7)

some Handlers have public Context while some have private context, what is the rule?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/PostbuildPhaseHandler.cs]: i should make all of them public

---
In reply to: [89707591](https://github.com/dotnet/docfx/pull/929#discussion_r89707591) [](ancestors = 89707591)

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs (hunk 1 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: nice, see that

---
In reply to: [89715562](https://github.com/dotnet/docfx/pull/929#discussion_r89715562) [](ancestors = 89715562,89707493)

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**